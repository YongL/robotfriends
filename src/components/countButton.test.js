import { shallow } from 'enzyme';
// import { shallow, mount, render, configure } from 'enzyme';
import React from 'react'; // import this because it is jsx
import CountButton from './countButton';


// import Adapter from 'enzyme-adapter-react-16';

// configure({adapter: new Adapter()});

// the test script come with react script /// create-react-app 


// shallow only shollow or render one give component it doesn't care anything inside or children
// inside it
// the document doesn't really give a specific direction of where to 
// to put the configure// 
// under src create setupTests.js 
// https://airbnb.io/enzyme/docs/installation/index.html
const color = 'red';
it('render CountButton', () => expect(shallow(<CountButton />).length).toBe(1));

it('snapshots CountButton', () => expect(shallow(<CountButton color={color}/>)).toMatchSnapshot());

describe('correctly increment counter', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CountButton color={color}/>);
    });

    it('increment', () => {
        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({count: 1});
    })

    it('has red color', () => {
        expect(wrapper.instance().props.color).toBe('red');
    });

    // https://airbnb.io/enzyme/docs/api/ShallowWrapper/props.html
    // check instance when use new enzym
})

// jest do snaphot testing, which snapshot the codes (string copy not image)

