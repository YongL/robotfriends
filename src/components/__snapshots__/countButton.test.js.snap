// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots CountButton 1`] = `
<div>
  <button
    className="f1"
    color="red"
    id="counter"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className="f1"
    onClick={[Function]}
  >
    update to 3
  </button>
</div>
`;
